examples/swarm/test_db.py

```diff
@@ -19,41 +19,101 @@
-port = int(Configuration.get('nupic.cluster.database.port'))
-user = Configuration.get('nupic.cluster.database.user')
-passwd = Configuration.get('nupic.cluster.database.passwd')
-
-print 
-print "This script will validate that your MySQL is setup correctly for NuPIC."
-print "MySQL is required for NuPIC swarming. The settings are defined in a "
-print "configuration file found in $NUPIC/src/nupic/support/nupic-default.xml "
-print "Out of the box those settings contain MySQL's default access "
-print "credentials."
-print
-print "The nupic-default.xml can be duplicated to define user specific changes, "
-print "calling the copied file $NUPIC/src/nupic/support/nupic-site.xml"
-print "Refer to the nupic-default.xml for additional instructions."
-print
-print "Defaults: localhost, 3306, root, no password"
-print
-print "Retrieved the following settings from NuPIC configuration:"
-print "    host   :    ", host
-print "    port   :    ", port
-print "    user   :    ", user
-print "    passwd :    ", passwd
-print
-
-try:
-  conn = pymysql.connect(host=host, port=port, user=user, passwd=passwd)
-except:
-  raise RuntimeError("Couldn't connect to the database."
-                     " Please ensure you have MySQL\n installed, running, and"
-                     " accessible using the NuPIC configuration settings.")
-
-print "Connection successful!!"
+DEFAULT_CONFIG = "nupic-default.xml"
+USER_CONFIG = "nupic-site.xml"
+
+
+def get_file_used():
+    """
+      Determine which NuPIC configuration file is being used and returns the
+      name of the configuration file it is using. Either DEFAULT_CONFIG or
+      USER_CONFIG.
+    """
+
+    # output will be {} if the file passed into Configuration._readConfigFile
+    # can not be found in the standard paths returned by
+    # Configuration._getConfigPaths.
+    output = Configuration._readConfigFile(USER_CONFIG) #pylint: disable=protected-access
+    if output != {}:
+        return USER_CONFIG
+    return DEFAULT_CONFIG
+
+def test_db_connection(host, port, user, passwd):
+    """
+      Determine if the specified host, port, user, passwd is able to connect
+      to a running mysql database, create test database, create a test table,
+      insert something into the table, delete the table, and delete the database.
+      returns true if this is successful, false if there is an error in this
+      process.
+    """
+    try:
+        conn = pymysql.connect(host=host, port=port, user=user, passwd=passwd)
+        cursor = conn.cursor()
+        cursor.execute('CREATE DATABASE IF NOT EXISTS testing')
+        conn.select_db('testing')
+        cursor.execute("CREATE TABLE test \
+                                (teststring VARCHAR(255),\
+                                 someint INT)")
+        cursor.execute("INSERT INTO test VALUES ('testing123', 123)")
+        cursor.execute('DROP TABLE IF EXISTS test')
+        cursor.execute('DROP DATABASE IF EXISTS testing')
```

Comment:
From [style guide](https://github.com/numenta/nupic/wiki/Python-Style-Guide#quotes); 

> By default, double quotes should be used for all strings.

Here and elsewhere throughout the file.